torch/testing/_internal/common_fsdp.py

```diff
@@ -823,7 +823,12 @@ def setUp(self):
 
     @property
     def world_size(self):
-        return torch.cuda.device_count() if torch.cuda.is_available() else 4
+        nranks = 4
+        if torch.cuda.is_available:
+            nranks = torch.cuda.device_count()
+        if torch.version.hip and nranks > 8:
```

Comment:
Why should this change be HIP-specific?

Reply:
@jithunnair-amd you are right, this is applicable to CUDA as well as long as there is a node with more than 8 GPUs.

Reply:
In that case, I'd suggest the following patch:
```
- return torch.cuda.device_count() if torch.cuda.is_available() else 4
+ return min(torch.cuda.device_count(), 8) if torch.cuda.is_available() else 4
```
And let's file an upstream PR with the same patch, with the description containing what you explained in https://github.com/ROCmSoftwarePlatform/pytorch/pull/1234#issuecomment-1580213588